src/Action/SetObjectFieldValueAction.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Action;
+
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\AdminBundle\Twig\Extension\SonataAdminExtension;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\PropertyAccess\PropertyPath;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+use Twig\Environment;
+
+final class SetObjectFieldValueAction
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var Environment
+     */
+    private $twig;
+
+    /**
+     * @var ValidatorInterface
+     */
+    private $validator;
+
+    public function __construct(Environment $twig, Pool $pool, $validator)
+    {
+        // NEXT_MAJOR: Move ValidatorInterface check to method signature
+        if (!($validator instanceof ValidatorInterface)) {
+            throw new \InvalidArgumentException(
+                'Argument 3 is an instance of '.get_class($validator).', expecting an instance of'
+                .' \Symfony\Component\Validator\Validator\ValidatorInterface'
+            );
+        }
+        $this->pool = $pool;
+        $this->twig = $twig;
+        $this->validator = $validator;
+    }
+
+    /**
+     * @throws NotFoundHttpException
+     *
+     * @return Response
+     */
+    public function __invoke(Request $request)
+    {
+        $field = $request->get('field');
```

Comment:
This method is very very long.... Can you try to split this?

Reply:
I know, but I don't see an easy way to do that, and doing so would be out of scope of this PR, which is already very big.